File: atlas/examples/kotlin/domain/src/main/kotlin/dev/atlas/example/domain/catalog/CatalogTitlePolicy.kt
package dev.atlas.example.domain.catalog

/**
 * Normalizes human-entered titles without exposing persistence or presentation concerns.
 */
interface CatalogTitlePolicy {
    /**
     * Produces a non-empty display title from an untrusted candidate.
     *
     * @param candidate Title text that may contain inconsistent spacing or casing.
     * @return Normalized title suitable for a catalog item.
     */
    fun normalize(candidate: String): String
}

File: atlas/examples/typescript/packages/domain/src/catalog/CatalogTitlePolicy.ts
/**
 * Normalizes human-entered titles without exposing a persistence or presentation concern.
 */
export interface CatalogTitlePolicy {
  /**
   * Produces a non-empty display title from an untrusted candidate.
   *
   * @param candidate - Title text that may contain inconsistent spacing or casing.
   * @returns Normalized title suitable for a catalog item.
   */
  normalize(candidate: string): string;
}
